mission/chapter02/react-todo-list/src/components/Section.tsx

```diff
@@ -0,0 +1,30 @@
+import { useDark } from "../context/darkProvider";
+import { TTodo } from "../types/todo";
+
+interface SectionProps {
+    title: string;
+    todos: TTodo[];
+    onClick?: (todo: TTodo) => void;
+    backgroundColor: string;
+    button_name: string;
+};
+// Map 사용 시, Key prop을 전달해야함, 리액트가 Key를 기준으로 rendering이 이뤄지기 때문이다!
+const Section = ({title, todos, onClick, backgroundColor, button_name}: SectionProps) => {
+    const { isDark } = useDark();
+
+  return (
+    <div className='render-container__section'>
+        <h2 className="render-container__title" style={{color: isDark? 'white' : 'black'}}>{title}</h2>
+        <ul id='todo-list' className="render-container__list">
+            {todos.map((todo) =>(
+                <li key={todo.id} className="render-container__item" style={{backgroundColor: isDark? '#3c3c3c' : '#f9f9f9'}}>
+                    <span className="render-container__item-text" style={{color: isDark? 'white' : 'black'}}>{todo.text}</span>
```

Comment:
todo.text와 같이 객체의 프로퍼티를 접근할 때 
```suggestion
    const {id, text} = todo;
```
로 먼저 비구조화 시킨 후에 넣는 방법도 제안드립니당

Reply:
비구조화를 먼저 시키려면 map 내부에서 return 구문이 필요하더군요! 또 다른 배움을 얻었습니다~!